### PR TITLE
WIP GRAPH 3: Custom Shapes and Colors

### PR DESCRIPTION
**Pull request by arniebradfo**
## Description ##
This adds the option to select from a set of polygon representations for each beacon node, and add color to all nodes. see screenshots.

## TODOs ##
- [x] Add mobx and server state state for saving custom shape and className data for nodes
- [x] Add graph styles once nodes can be customized

## Testing ##
...

## Screenshots ##
Screen Shot 2023-04-07 at 12 01 48 PM
Screen Shot 2023-04-07 at 12 02 03 PM
Screen Shot 2023-04-07 at 12 02 31 PM
Screen Shot 2023-04-07 at 12 02 34 PM
Screen Shot 2023-04-07 at 12 03 00 PM
Screen Shot 2023-03-31 at 5 49 32 PM
Screen Shot 2023-03-31 at 5 50 00 PM

### CODE REVIEW

**Comment by GoldingAustin** [comment]
@arniebradfo I added shapes and colors to the beacon, host, and server meta-objects. They can be updated in their meta mutations

File: packages/cs-parser/src/beacon/parsingOrchestrator.machine.ts [hunk 1]
```diff
@@ -141,6 +151,7 @@ export const parsingOrchestratorMachine = createMachine(
 								ip,
 								os: findOsFromMetaLine(metadataLog.blob),
 								host: metadataLog.beacon?.host as Host,
+								shape: Shapes.circle,
```
Comment (arniebradfo) [inline packages/cs-parser/src/beacon/parsingOrchestrator.machine.ts:154]: @GoldingAustin I had to add `shape: Shapes.circle` here in `hostMeta` and in `beaconMeta`. Not sure why its marked as required?

File: applications/server/src/store/host-resolvers.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export class HostResolvers {
 		@Arg('hostDisplayName', () => String) hostDisplayName: string,
 		@Arg('shape', { nullable: true }) shape?: Shapes,
 		@Arg('color', { nullable: true }) color?: string,
-		@RelationPath() relationPaths: Relation<Host>
+		@RelationPath() relationPaths?: Relation<Host>
```
Comment (arniebradfo) [inline applications/server/src/store/host-resolvers.ts:34]: @GoldingAustin, This was giving me an error that `A required parameter cannot follow an optional parameter. ts(1016)` so I made it optional?

File: applications/server/schema.graphql [hunk 3]
```diff
@@ -483,7 +483,7 @@ type Query {
   logsByBeaconId(beaconId: String!, campaignId: String!): [LogEntry!]!
 
   """"""
-  nonHidableEntities(beaconIds: [String!] = [], campaignId: String!, hostIds: [String!] = []): NonHidableEntities!
+  nonHidableEntities(beaconIds: [String!]! = [], campaignId: String!, hostIds: [String!]! = []): NonHidableEntities!
```
Comment (arniebradfo) [inline applications/server/schema.graphql:486]: @GoldingAustin, this just updated itself on build

**Comment by arniebradfo** [comment]
@GoldingAustin & @sharplessHQ, I think I did an okay job merging in develop 489b1d9, but some of the file changes in `Beacon.tsx` and `Host.tsx` were not automatically carried over because git did not track the file rename to `BeaconRow.tsx` and `HostOrServerRow.tsx` respectively. I copied these over manually, but it was hard to know if I merged them correctly. This may need a review.

**Review by GoldingAustin — APPROVED** [review]

**Pull request merged**